crates/oxc_linter/src/rules/eslint/func_names.rs

```diff
@@ -0,0 +1,507 @@
+use oxc_ast::ast::{
+    AssignmentTarget, AssignmentTargetProperty, BindingPatternKind, Expression, Function,
+    FunctionType, PropertyKind,
+};
+use oxc_ast::AstKind;
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::{Atom, Span};
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Default, Clone)]
+pub struct FuncNames {
+    default_config: FuncNamesConfig,
+    generators_config: FuncNamesConfig,
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+enum FuncNamesConfig {
+    #[default]
+    Always,
+    AsNeeded,
+    Never,
+}
+
+impl FuncNamesConfig {
+    pub fn from(raw: &str) -> Self {
+        match raw {
+            "always" => FuncNamesConfig::Always,
+            "as-needed" => FuncNamesConfig::AsNeeded,
+            "never" => FuncNamesConfig::Never,
+            _ => FuncNamesConfig::default(),
+        }
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Require or disallow named function expressions
+    ///
+    /// ### Why is this bad?
+    ///
+    /// If you leave off the function name then when the function throws an exception you are likely
+    /// to get something similar to anonymous function in the stack trace. If you provide the optional
+    /// name for a function expression then you will get the name of the function expression in the stack trace.
+    ///
+    /// /// ### Example
+    ///
+    /// ```javascript
+    /// Foo.prototype.bar = function bar() {};
```

Comment:
Please provide good/bad examples and indicate which one is which. See the docs for no-unused-vars for an example.

Reply:
https://github.com/oxc-project/oxc/blob/6ff200d072fe55cb720c8dbe3a00f885fd1bbd28/crates/oxc_linter/src/rules/eslint/no_unused_vars/mod.rs#L67

Notice how examples are given based on the configuration provided.

I plan on using these doc comments to build documentation pages on oxc.rs in the near future, fleshed out documentation will help with DX.

Reply:
I updated the doc, feel free to modify it as needed :)